src/java/voldemort/client/ClientConfig.java

```diff
@@ -571,6 +577,52 @@ public ClientConfig setSocketTimeout(int socketTimeout, TimeUnit unit) {
         return this;
     }
 
+    public long getIdleConnectionTimeout(TimeUnit unit) {
+        if (idleConnectionTimeoutMs <= 0) {
+            return idleConnectionTimeoutMs;
+        }
+        return unit.convert(idleConnectionTimeoutMs, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Set the timeout for idle connections. Voldemort client caches all
+     * connections to the Voldemort server. This setting allows the a connection
+     * to be dropped, if it is idle for more than this time.
+     * 
+     * This could be useful in the following cases 1) Voldemort client is not
+     * directly connected to the server and is connected via proxy of firewall.
+     * The Proxy or firewall could drop the connection silently. If the
+     * connection is dropped, then client will see operations fail with a
+     * timeout. Setting this property enables the Voldemort client to tear down
+     * the connection before a firewall could drop it. 2) Voldemort server
+     * caches the connection and each connection has an associated memory cost.
+     * Setting this property on all clients, enable the clients to prune the
+     * connections and there by freeing up the server connections.
+     * 
+     * throws IllegalArgumentException if the timeout is less than 10 minutes.
+     * 
+     * @param idleConnectionTimeout
+     *            zero or negative number to disable the feature ( default -1)
+     *            timeout
+     * @param unit
+     *            {@link TimeUnit}
+     * @return ClientConfig object for chained set
+     * 
+     *         throws {@link IllegalArgumentException} if the timeout is greater
+     *         than 0, but less than 10 minutes.
+     */
+    public ClientConfig setIdleConnectionTimeout(long idleConnectionTimeout, TimeUnit unit) {
```

Comment:
Thanks for the detailed JavaDoc. Can we add a line mentioning that this config is only (potentially) relevant to very low throughput clients? Also, a small comment on why the 10 minutes minimum would be great.